### PR TITLE
Fix Conflict Detection

### PR DESCRIPTION
**Pull request by lkiesow**
This patch fixes Opencast's conflict detection for scheduled events
which would marks two events (9am to 10am) and (10am to 11am) as
conflicting which is a very exexpected behavior.

Worse, even disjoind end/start times could cause conflicts with no sane
reason why they would. E.g. (8:00:00 - 9:00:00) and (9:00:10 to 9:30:00)
would be marked as conflicting.

This patchfixes this erratic behavior, returning to the iCal definition
for start dates, end dates and conflicts we have always used.

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] be against the correct branch (features can only go into develop)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated testing
* [ ] have a clean commit history
* [ ] have proper commit messages (title and body) for all commits
* [ ] have appropriate tags applied

### CODE REVIEW

**Comment by smarquard** [comment]
It doesn't make sense to just set EVENT_MINIMUM_SEPARATION_MILLISECONDS = 0 because that's a concept used in the code e.g. in eventWithinMinimumSeparation() so setting it to 0 defeats the whole purpose.

Functionally, I think many capture agents are going to have difficulty dealing with consecutive recordings with no interval between them. So this seems like a breaking change as far as CAs are concerned. If Opencast permits this, then CAs need to understand to stop one recording early or start the next one late if there's time required (as there typically is) to shutdown a recording pipeline, move files around,  and start a new one.

At least this should be configurable with a safe default, and production sites can set it to 0 at their own risk.

**Comment by lkiesow** [comment]
This is actually a regression bug since it was definitely possible to configure this before. There is also no documentation for capture agents anywhere which guarantees capture agents any pause between any recordings, so it should not break any reasonable implementation. As a capture agent author, I can also say that this is no problem for pyCA.

Setting this to 0 was the minimal change for me to fix this bug. But I can make this configurable if you would like me to.

**Comment by smarquard** [comment]
Firstly, I'm happy with the change if it's configurable. I think the default should be the same as present (60s) but don't mind too strongly about that. If the default is not the same, it could be a breaking change for some sites and CAs, and should be listed in the changelog.

For reference, I checked where this was introduced and it was part of a set of changes to the scheduler code in MH-12709 PR
https://github.com/opencast/opencast/pull/112

The change was clearly intentional and reviewed, so I don't think it's appropriate to call this a regression especially as it's been like since 4.2 it seems. I appreciate you have a different view of what correct behaviour is but "as it's always been" is now quite a few releases ago.

**Comment by gregorydlogan** [comment]
Back in the day there was definitely a 1 minute minimum separation between recordings.  This was in the code, and tested, but got lost by the way side.  It was a byproduct of the hardware at the time not being able to cycle over to a new capture in (reliably) less than 30 secondss.

**Comment by gregorydlogan** [comment]
Looking back at the logs, the minimum appears to have been added in 4.x as part of https://opencast.jira.com/browse/MH-12709, with 31d6d16b04d.  Presumably there was a good reason for that.  Whether it is still valid today is unclear.

If we're going to remove that separation, let's get rid of the variable entirely.  No sense in keeping dead vars around.  If we're not, then let's document it.  IMO, I would document that one minute turnaround and be done with it.

**Comment by gregorydlogan** [comment]
I'm going to close this.  If you've still got a customer complaining about this we can reopen and continue discussion.

**Comment by Rillke** [comment]
>  If you've still got a customer complaining about this we can reopen and continue discussion.

Though it's not high-priority. 60s is a lot, according to our lecturers  (that's 16 times 2 times 1 minute = about 1/3 lecture over a semester). If the buffer time would be either configurable, or removed, they'd be happy. Currently I have to recompile Opencast in order to get rid of this and end a discussion which took  place amongst 4 well-paid people for 20 minutes.

Our Stud.IP plugin takes care of the scheduling and has a configurable buffer time built-in.

**Comment by Rillke** [comment]
[Here's the full reasoning for that.](https://groups.google.com/a/opencast.org/g/users/c/FvAaw_5TUoA/m/3Fv98pvABQAJ) In case it's rejected, it's also not too bad for us, as we could either continue to compile a custom version of opencast, have a configurable buffer time, or tell our room service that for technical reason, they have to change their current practice.

**Comment by gregorydlogan** [comment]
So, now that the original Matterhorn source is posted, I can finally go digging properly.  The code I'm remembering was added by https://github.com/opencast/matterhorn/commit/9f59cf8f4a84d050da18658bb6cae454aef8d805.  I can't find any enforcement in the admin UI, but the code is truly spaghetti (there's a reason it was replaced!) and I distinctly remember the UI not accepting sub-minute turnaround times.

That being said, the reason for adding this is *12 years* old.  I'm not saying it has to stay, since it's not technically part of the CA API, but hardware has come a long way.  If we're seeing a customer need to make this configurable, or go away, I'm fine with it.  Personally, I would remove it entirely and leave enforcement of any kind of conflict detection to customer business rules.  In large part, this is because we don't know what hardware you have - your admin staff need to not schedule things which won't work with your hardware.

tl;dr: Fine, remove these checks.  But make sure there's something in the release notes to make it clear that your mileage may vary if you schedule things to start and stop at the same time.

**Comment by geichelberger** [comment]
Can this PR be part of OC 12?

**Comment by lkiesow** [comment]
Switched the branch.
Now the only thing left is for someone to review this.

**Review by geichelberger — APPROVED** [review]
LGTM had to rebase it locally before testing, but that should not make any difference.

**Comment by gregorydlogan** [comment]
@lkiesow Can you ensure there is a note in the release notes for this?
